src/include/homestore/homestore_decl.hpp

```diff
@@ -138,11 +140,16 @@ struct hs_input_params {
     io_flag data_open_flags{io_flag::DIRECT_IO}; // All data drives open flags
     io_flag fast_open_flags{io_flag::DIRECT_IO}; // All index drives open flags
 
+
     uint64_t app_mem_size{static_cast< uint64_t >(1024) * static_cast< uint64_t >(1024) *
                           static_cast< uint64_t >(1024)}; // memory available for the app (including cache)
     uint64_t hugepage_size{0};                            // memory available for the hugepage
     bool is_read_only{false};                             // Is read only
     bool auto_recovery{true};                             // Recovery of data is automatic or controlled by the caller
+    ChunkSelector* dataChunkSelector{nullptr};        // ChunkSelector for DataService
+    ChunkSelector* logChunkSelector{nullptr};          // ChunkSelector for LogService
```

Comment:
I think ChunkSelector will be used by data service only, there is no need to provide a chunkselector for other services.

Reply:
chunkSelector will be a parameter of the constructor of vDev, and every service will have a vDev, so we need to provide a chunkSelector for every service. for dataService , we can pass the actual chunkSelector(heapChunkSelector)  provided by homeObject. for other three service, we could just pass nullptr to ignore chunkSelector.

the default value of all the chunkSelectors here are nullptr.  the code here does not take much modifications while leaving a simple port for other service to implement their own chunkSelector if necessary in the future. it unifies the process of all the services to create a vDev